src/components/common/Extensions/ExtensionDetails/tableData.tsx

```diff
@@ -0,0 +1,120 @@
+import React from 'react';
+import { MessageDescriptor, defineMessages, FormattedDate } from 'react-intl';
+
+import { AnyExtensionData, Colony } from '~types';
+import { isInstalledExtensionData } from '~utils/extensions';
+import DetailsWidgetUser from '~shared/DetailsWidgetUser';
+import InvisibleCopyableAddress from '~shared/InvisibleCopyableAddress';
+import MaskedAddress from '~shared/MaskedAddress';
+
+import ExtensionStatusBadge from '../ExtensionStatusBadge';
+import { displayName } from './ExtensionDetailsAside';
+
+import styles from './ExtensionDetails.css';
+
+const MSG = defineMessages({
+  status: {
+    id: `${displayName}.status`,
+    defaultMessage: 'Status',
+  },
+  installedBy: {
+    id: `${displayName}.installedBy`,
+    defaultMessage: 'Installed by',
+  },
+  dateInstalled: {
+    id: `${displayName}.dateInstalled`,
+    defaultMessage: 'Date installed',
+  },
+  dateCreated: {
+    id: `${displayName}.dateCreated`,
+    defaultMessage: 'Date created',
+  },
+  latestVersion: {
+    id: `${displayName}.latestVersion`,
+    defaultMessage: 'Latest version',
+  },
+  developer: {
+    id: `${displayName}.developer`,
+    defaultMessage: 'Developer',
+  },
+  versionInstalled: {
+    id: `${displayName}.versionInstalled`,
+    defaultMessage: 'Version installed',
+  },
+  contractAddress: {
+    id: `${displayName}.contractAddress`,
+    defaultMessage: 'Contract address',
+  },
+});
+
+export type TableRowData = {
+  label: MessageDescriptor;
+  value: string | JSX.Element;
+};
+
+export const getTableData = (
```

Comment:
I split it into two functions `get(Installed/Installable)ExtensionTableData` since they need access to the extension data and colony. Then in `getTableData` I call the appropriate function. Hope this is closer to what you had in mind